rlgym_tools/extra_state_setters/goalie_state.py

```diff
@@ -0,0 +1,164 @@
+        :param reset_to_max_boost: Boolean indicating whether the cars will start each episode with 100 boost or keep from last episode
+        """
+        super().__init__()
+        self.team_turn = 0 #swap every reset who's getting shot at
+        
+        self.aerial_only = aerial_only
+        self.allow_enemy_interference = allow_enemy_interference
+        self.first_defender_in_goal = first_defender_in_goal
+        self.reset_to_max_boost = reset_to_max_boost
+
+    def reset(self, state_wrapper: StateWrapper):
+        """
+        Modifies the StateWrapper to set a new shot
+
+        :param state_wrapper: StateWrapper object to be modified with desired state values.
+        """
+        self._reset_ball(state_wrapper, self.team_turn, self.aerial_only)
+        self._reset_cars(state_wrapper, self.team_turn, self.first_defender_in_goal, self.allow_enemy_interference, self.reset_to_max_boost)
+        
+        #which team will recieve the next incoming shot
+        self.team_turn = (self.team_turn + 1) % 2
+    
+
+    def _reset_cars(self, state_wrapper: StateWrapper, team_turn, first_defender_in_goal, allow_enemy_interference, reset_to_max_boost):
+        """
+        Function to set cars in preparation for an incoming shot
+
+        :param state_wrapper: StateWrapper object to be modified.
+        :param team_turn: team who's getting shot at
+        :param allow_enemy_interference: Boolean indicating whether opponents will spawn close enough to easily affect the play
+        :param first_defender_in_goal: Boolean indicating whether the first defender will spawn in the goal
+        :param reset_to_max_boost: Boolean indicating whether the cars will start each episode with 100 boost or keep from last episode
+        """
+        first_set = False
+        for car in state_wrapper.cars:
+            # set random position and rotation for all cars based on pre-determined ranges
+            
+            if car.team_num == team_turn and not first_set:   
+                if first_defender_in_goal:
+                    y_pos = -GOAL_LINE if car.team_num == 0 else GOAL_LINE
+                    car.set_pos(0, y_pos, z=17)
+                    first_set = True
+                else:
+                    self._place_car_in_box_area(car, team_turn)
+                    
+            else:
+                if allow_enemy_interference:
+                    self._place_car_in_box_area(car, team_turn)
+                
+                else:
+                    self._place_car_in_box_area(car, car.team_num)                    
+            
+            
+            car.set_rot(0, rand.random() * YAW_MAX - YAW_MAX/2, 0) 
+            
+            if reset_to_max_boost:
+                car.boost = 100
+            
+    def _place_car_in_box_area(self, car, team_delin):
+        """
+        Function to set a new ball towards a goal
```

Comment:
This should probably say something different. Maybe 'Places given car into the area of a predetrmined box or something along those lines'